src/Sulu/Bundle/SecurityBundle/Entity/UserRole.php

```diff
@@ -124,6 +124,15 @@ public function getUser()
      */
     public function setRole(RoleInterface $role)
     {
+        if ($role->getAnonymous()) {
+            throw new \LogicException(
```

Comment:
Two questions here:

1. If it is not allowed to add an anonymous role to a user, does that mean there will be no anonymous user? I.e. that in the code where we need to check for anonymous, we are going to just load the anonymous role instead of getting the roles from a user object?
2. Shouldn't we introduce a more specific exception for that? Could imagine that the calling code might try to recover from that specific error. Would think about something like a `AssignAnonymousRoleException`

Reply:
1. Yes I would think that there should be no Anonymous User. As I would think that would break also currently how we did handle anonymous User in project by just checking for `if (!$user instanceof SuluUserInterface)`

2. Sounds good for me 👍